### PR TITLE
WIP: Added some missing field info to the EngineApi interface

### PR DESCRIPTION
**Pull request by lukaspj**
This probably doesn't have any effect on how the engine works, but it is nice for documentation and engine API export purposes.

### CODE REVIEW

File: Engine/source/core/util/tVector.h [hunk 1]
```diff
@@ -63,11 +63,12 @@ extern bool VectorResize(U32 *aSize, U32 *aCount, void **arrayPtr, U32 newCount,
 template<class T>
 class Vector
 {
-  protected:
+public:
```
Comment (JeffProgrammer) [inline Engine/source/core/util/tVector.h:66]: quite dangerous, completely breaks encapsulation...

Comment (lukaspj) [reply Engine/source/core/util/tVector.h:66]: Yes I forgot I made that change in there, wanted to bring it up and ask if there were another way to do it.. Maybe moving the Implement Struct to somewhere associated with Vector?

**Comment by lukaspj** [comment]
Added WIP tag, until we can figure out how to handle Vector stuff.

File: Engine/source/core/util/tVector.h (hunk 1 above)
Comment (lukaspj) [reply Engine/source/core/util/tVector.h:66]: An alternative solution is hard-coding the offsets:
```c++
IMPLEMENT_STRUCT( Vector< bool >,
   BoolVector,,
   "" )
   {"elementCount", "", 1, TYPE<U32>(), 0},
   {"arraySize", "", 1, TYPE<U32>(), 4},
   {"array", "", 1, TYPE<bool*>(), 8},
   //FIELD(mElementCount, elementCount, 1, "")
   //FIELD(mArraySize, arraySize, 1, "")
   //FIELD(mArray, elements, 1, "")
END_IMPLEMENT_STRUCT;
```

Comment (lukaspj) [reply Engine/source/core/util/tVector.h:66]: Or the offsets could be computed and defined in tVector.h, but that sort of breaks encapsulation as well.

**Comment by lukaspj** [comment]
Structs that have access permission issues:

* Vectors (IntVector, FloatVector, BoolVector)
* UUID
* MatrixF

**Comment by lukaspj** [comment]
Closed in favor of #2366

**Pull request closed without merging**